test/jdk/java/awt/Focus/UnaccessibleChoice/AccessibleChoiceTest.java

```diff
@@ -23,115 +23,128 @@
-    Window win = new Window(frame);
-    Choice choice = new Choice();
-    Button def = new Button("default owner");
-    CountDownLatch go = new CountDownLatch(1);
-
-    public static void main(final String[] args) throws IOException {
-        AccessibleChoiceTest app = new AccessibleChoiceTest();
-        app.test();
-    }
+    static Frame frame = new Frame("Accessible Choice Test Frame");
+    static Choice choice = new Choice();
+    static Button button = new Button("default owner");
+    static CountDownLatch go = new CountDownLatch(1);
+    static volatile Point loc;
 
-    private void test() throws IOException {
+    public static void main(final String[] args) throws Exception {
         try {
-            init();
-            start();
+            System.out.println("ONE");
+            createAndShowUI();
+            System.out.println("TWO");
+            test();
+            System.out.println("THREE");
         } finally {
-            if (frame != null) frame.dispose();
-            if (win != null) win.dispose();
+            if (frame != null) {
+                System.out.println("FOUR");
+                frame.dispose();
+            }
         }
     }
 
-    public void init() {
-        win.setLayout (new FlowLayout ());
-        win.add(def);
-        def.addFocusListener(new FocusAdapter() {
+    public static void createAndShowUI() throws Exception {
+        EventQueue.invokeAndWait(() -> {
+            System.out.println("FIVE");
+            frame.setLayout(new FlowLayout());
+            frame.add(button);
+            button.addFocusListener(new FocusAdapter() {
                 public void focusGained(FocusEvent e) {
                     go.countDown();
                 }
             });
-        choice.add("One");
-        choice.add("Two");
-        win.add(choice);
+            choice.add("One");
+            choice.add("Two");
+            frame.add(choice);
+            frame.pack();
+            frame.setLocationRelativeTo(null);
+            frame.setVisible(true);
+        });
     }
 
-    public void start () throws IOException {
-        frame.setVisible(true);
-        win.pack();
-        win.setLocation(100, 200);
-        win.setVisible(true);
-
-        Robot robot = null;
+    public static void test() throws Exception {
+        Robot robot;
         try {
             robot = new Robot();
         } catch (Exception ex) {
             throw new RuntimeException("Can't create robot");
         }
         robot.waitForIdle();
         robot.delay(1000);
-        robot.setAutoDelay(150);
         robot.setAutoWaitForIdle(true);
 
+        System.out.println("SIX");
+
         // Focus default button and wait till it gets focus
-        Point loc = def.getLocationOnScreen();
-        robot.mouseMove(loc.x+2, loc.y+2);
+        EventQueue.invokeAndWait(() -> {
+            loc = button.getLocationOnScreen();
+        });
+        robot.mouseMove(loc.x + button.getWidth() / 2, loc.y
```

Comment:
get button's width and height on EDT along with location.

Reply:
Done